### DISCUSSION TITLE
Video Playback speed support?

### DISCUSSION THREAD

**Discussion by radubirsan** [airsdk/Adobe-Runtime-Support#1628] [Ideas]
HTML supports playback speed modification, will AIR support a similar feature?

**Comment by ajwfrost** [comment]
This would be the plan for our 'new' AIR multimedia APIs. Assuming there's support for this from the operating system functionality that underpins it - so I'm pretty sure I've seen this using the Windows implementation..

**Reply by VCone** [reply]
Any updates on the AIR Multimedia API's?

If taking requests for features... 
(1) Please let it **encode H.264** (all profiles/levels, if possible, but even only Baseline profile is acceptable).
(2) Please let it **decode AAC into PCM** (encoding **from PCM back into AAC** would be a nice feature too).

This would allow for dynamic (audio/video) content creation. For example :
- Using Audio's SampleData with decoded AAC we could now adjust the audio's pitch during the adjusted video speed. 
- Video-input gets visual effects edited, before it's re-streamed as live video.
_eg_:  digital score overlaid on some sports video, or motion tracked web-cam effects, etc.
- Video editing applications (_eg:_ apply colour LUTs to some new output video).
- Converting BitmapData-based effects into video (could be video vfx, could be educational diagrams/simulations,, etc)

Most SDK's have the power to encode (see: Android / Swift / C# / Python).
Chrome browser has the WebCodecs API. Would it be possible to leverage WebCodecs somehow into AS3's Webkit engine?

Thanks.

**Comment by markkelynack** [comment]
Please, please add in a loop video boolean, that would be so sooo good.
